src/Core/AppleNative/PlatformInterop/PlatformInterop/CALayer+AutoresizingMask.m

```diff
@@ -0,0 +1,73 @@
+#import <objc/runtime.h>
+#import <UIKit/UIKit.h>
+#import "CALayer+AutoresizingMask.h"
+
+@implementation CALayer (AutoresizingMask)
+
+static char CALayerAutoresizingMask;
+
++ (void)load
+{
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        Method originalSetBounds = class_getInstanceMethod(self, @selector(setBounds:));
+        Method swizzledSetBounds = class_getInstanceMethod(self, @selector(maui_setBounds:));
+        method_exchangeImplementations(originalSetBounds, swizzledSetBounds);
+    });
+}
+
+- (void)maui_setBounds:(CGRect)bounds
+{
+    // Only if size has changed
+    if (!CGSizeEqualToSize(self.bounds.size, bounds.size))
+        for (CALayer *layer in self.sublayers)
```

Comment:
TODO: can we avoid this loop by somehow storing the count of sublayers with an autoresize mask?